lib/r10k/source/git.rb

```diff
@@ -113,7 +119,17 @@ def desired_contents
 
   def branch_names
     opts = {:prefix => @prefix, :invalid => @invalid_branches, :source => @name}
-    @cache.branches.map do |branch|
+    branches = @cache.branches
+    if !@branch_filter.nil? && !@branch_filter.empty?
+      branches = branches.select do |branch|
```

Comment:
The select call could be lifted into its own method and then unit tested on its own. `filter_branches` or something of the sort.

Reply:
Workin' on that.

Reply:
Well, I lifted it out, but it's a private method, as is the branch_names method. Should we be unit testing the private methods? I can, but I thought that was considered bad testing methodology.